qos_controller/serenity_controller.cpp

```diff
@@ -23,18 +22,25 @@ class SerenityControllerProcess :
     public Process<SerenityControllerProcess> {
  public:
   SerenityControllerProcess(
-      const lambda::function<Future<ResourceUsage>()>& _usage)
+      const lambda::function<Future<ResourceUsage>()> _usage)
       : usage(_usage) {}
 
   Future<list<QoSCorrection>> corrections() {
+    return this->usage()
+      .then(defer(self(), &Self::_corrections, lambda::_1));
+  }
+
+  Future<list<QoSCorrection>> _corrections(
+      const Future<ResourceUsage>& _resourceUsage) {
     // TODO(bplotka) Set up the main qos correction pipeline here.
     std::cout << "Serenity QoS Controller pipeline run." << "\n";
-    // For now return empty resources.
+
+    // For now return empty Future.
```

Comment:
Is this actually what you want?
Just checking that you don't mean an empty list?